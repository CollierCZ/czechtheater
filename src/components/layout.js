import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { StaticQuery, graphql } from 'gatsby'

import '../assets/scss/main.scss'
import logoWide from '../images/logo-wide.jpg'

const Layout = ({ children, location }) => {

  let content;

  if (location && location.pathname === '/') {
    content = (
      <div>
        {children}
      </div>
    )
  } else {
    content = (
      <div id="wrapper" className="page">
        <div>
          {children}
        </div>
      </div>
    )
  }

  return (
    <StaticQuery
      query={graphql`
        query SiteTitleQuery {
          site {
            siteMetadata {
              title
              description
              keywords
            }
          }
        }
      `}
      render={data => (
        <>
          <Helmet
            title={data.site.siteMetadata.title}
            meta={[
              { name: 'description', content: data.site.siteMetadata.description },
              { name: 'keywords', content: data.site.siteMetadata.keywords },
              { name: 'og:image', content: logoWide}
            ]}
          >
            <html lang="en" />
          </Helmet>
          {content}
        </>
      )}
    />
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
