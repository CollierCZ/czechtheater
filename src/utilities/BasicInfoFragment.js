import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KenticoCloudItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedHtml
            }
            favicon {
                assets{
                  url
                }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;


