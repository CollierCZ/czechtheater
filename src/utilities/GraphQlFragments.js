import { graphql } from 'gatsby';

export const BasicInfoFragment = graphql`
    fragment BasicInfoFragment on KontentItemBasicInfo {
        elements {
            about_us {
              value
            }
            contact_info {
              resolvedData {
                html
              }
            }
            favicon {
              value {
                url
              }
            }
            name {
              value
            }
            short_description {
              value
            }
            social_media {
              value
            }
          }
    }
`;

export const ShowListFragment = graphql`
    fragment ShowListFragment on KontentItemShowSection {
      elements {
        shows {
          linked_items {
            elements {
              main_image {
                value {
                  url
                  description
                }
              }
              name {
                value
              }
              url {
                value
              }
              short_description {
                value
              }
              ticket_link {
                value
              }
              gallery {
                value {
                  url
                  description
                }
              }
            }
          }
        }
      }
    }
`;


