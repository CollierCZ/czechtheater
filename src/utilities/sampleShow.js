export const sampleShow = {
    elements: {
      name: {
        value: "Alquist's revenge"
      },
      description: {
        resolvedHtml: "All the acting"
      },
      main_image: {
        assets: [{
          description: "A cool image",
          url: "https://example.com"
        }]
      },
      ticket_link: {
        value: "https://example.com"
      },
      url: {
        value: "alquists-revenge"
      },
      gallery: {
          assets: [
              {
                  description: "A cool image",
                  url: "https://example.com"
              }
          ]
      }
    }
  }