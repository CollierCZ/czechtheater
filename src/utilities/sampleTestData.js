export const sampleShow = {
  elements: {
    name: {
      value: "Alquist's revenge"
    },
    description: {
      resolvedHtml: "<p>All the acting</p>"
    },
    main_image: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    },
    ticket_link: {
      value: "https://example.com"
    },
    url: {
      value: "alquists-revenge"
    },
    gallery: {
      assets: [{
        description: "A cool image",
        url: "https://example.com"
      }]
    }
  }
}

export const sampleShowNoGallery = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.gallery.assets = []

export const sampleShowNoTicketLink = JSON.parse(JSON.stringify(sampleShow))
sampleShowNoGallery.ticket_link.value = ""

export const basicInfoQuery = {
  kenticoCloudItemBasicInfo: {
    elements: {
      name: {
        value: "Czech Theater"
      },
      favicon: {
        assets: [{
          url: "https://assets-us-01.kc-usercontent.com/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg"
        }]
      },
      short_description: {
        value: "A great theater."
      }
    }
  }
}