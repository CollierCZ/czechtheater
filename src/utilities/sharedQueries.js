import { graphql } from "gatsby";

export const basicInfoQuery = graphql`
  query BasicInfoQuery {
    kenticoCloudItemBasicInfo {
      elements {
        name {
          value
        }
        short_description {
          value
        }
        favicon {
          assets{
            url
          }
        }
      }
    }
  }
`