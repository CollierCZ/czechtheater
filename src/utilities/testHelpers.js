import { shallow } from "enzyme"

export const simpleShallowRender = (component) => {
  const wrapper = shallow (component)
  expect(wrapper).toMatchSnapshot()
}