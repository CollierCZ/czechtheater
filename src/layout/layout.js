import React from "react"
import PropTypes from "prop-types"
import { StaticQuery, graphql } from "gatsby"

import Header from "../components/Header/Header"
import "./layout.css"

const Layout = ({ children }) => (
  <StaticQuery
    query={graphql`
      query SiteTitleQuery {
        kenticoCloudItemBasicInfo {
          elements {
            name {
              value
            }
            favicon {
              assets {
                url
                description
              }
            }
          }
        }
      }
    `}
    render={data => (
      <>
        <Header 
          siteTitle={data.kenticoCloudItemBasicInfo.elements.name.value} 
          logoUrl={data.kenticoCloudItemBasicInfo.elements.favicon.assets[0].url} 
          logoDescription={data.kenticoCloudItemBasicInfo.elements.favicon.assets[0].description}        
        />
        <div
          style={{
            margin: `0 auto`,
            maxWidth: 960,
            padding: `0px 1.0875rem 1.45rem`,
            paddingTop: 0,
          }}
        >
          <main>{children}</main>
          <footer>
            © 2018&ndash;{new Date().getFullYear()} Czech Theater, z.s., IČO: 07245904. Theme inspired by <a href="https://html5up.net">HTML5 UP</a>. Built with <a href="https://www.gatsbyjs.org">Gatsby</a>.
          </footer>
        </div>
      </>
    )}
  />
)

Layout.propTypes = {
  children: PropTypes.node.isRequired,
}

export default Layout
