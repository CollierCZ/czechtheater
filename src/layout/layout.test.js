import { basicInfoQuery } from "../utilities/sampleTestData" 
import { PureLayout as Layout } from "./layout";
import React from "react";
import { simpleShallowRender } from "../utilities/testHelpers";

describe("Layout", () => {
    it("loads correctly", () => {
      simpleShallowRender(<Layout data={basicInfoQuery}><p>A test</p></Layout>)
    })
})