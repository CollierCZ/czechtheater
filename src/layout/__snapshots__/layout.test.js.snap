// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout loads correctly 1`] = `
<Fragment>
  <Header
    logoUrl="https://assets-us-01.kc-usercontent.com:443/fb599585-baec-0077-1624-981d3321fff0/4e43fc7c-ce8d-49fd-aba9-bdf908330220/logo.jpg?fm=webp&w=88&h=88"
    siteTitle="Czech Theater"
  />
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0px 1.0875rem 1.45rem",
        "paddingTop": 0,
      }
    }
  >
    <main>
      <p>
        A test
      </p>
    </main>
    <footer>
      © 2018–
      2021
       Czech Theater, z.s., IČO: 07245904. Theme inspired by 
      <a
        href="https://html5up.net"
      >
        HTML5 UP
      </a>
      . Powered by 
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
       and 
      <a
        href="https://kontent.ai"
      >
        Kentico Kontent
      </a>
      .
    </footer>
  </div>
</Fragment>
`;
